framework/test/unit/modules/random/endpoint.spec.ts

```diff
@@ -148,10 +158,265 @@ describe('RandomModuleEndpoint', () => {
+			const count = 1000;
+			const distance = 'distance';
+			context.params = { address, seed, count, distance };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				"Lisk validator found 1 error[s]:\nProperty '.distance' should be of type 'integer'",
+			);
+		});
+
+		it('should throw error when count is less than 1', async () => {
+			// Arrange
+			const { address } = genesisDelegates.delegates[0];
+			const seed = genesisDelegates.delegates[0].hashOnion.hashes[1];
+			const count = 0;
+			const distance = 1000;
+			context.params = { address, seed, count, distance };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				'Lisk validator found 1 error[s]:\nmust be >= 1',
+			);
+		});
+
+		it('should throw error when distance is less than 1', async () => {
+			// Arrange
+			const { address } = genesisDelegates.delegates[0];
+			const seed = genesisDelegates.delegates[0].hashOnion.hashes[1];
+			const count = 1000;
+			const distance = 0;
+			context.params = { address, seed, count, distance };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				'Lisk validator found 1 error[s]:\nmust be >= 1',
+			);
+		});
+	});
+
+	describe('getSeeds', () => {
+		let address: string;
+
+		beforeEach(async () => {
+			// Arrange
+			address = genesisDelegates.delegates[0].address;
+			const seed = genesisDelegates.delegates[0].hashOnion.hashes[1];
+			const count = 1000;
+			const distance = 10;
+
+			await randomEndpoint.setHashOnion({ ...context, params: { address, seed, count, distance } });
+		});
+
+		it('should return an array of seed objects', async () => {
+			// Act
+			const storedSeed = await randomEndpoint.getHashOnionSeeds(context);
+
+			// Assert
+			expect(storedSeed.seeds).toHaveLength(1);
+			expect(storedSeed.seeds[0]).toEqual({
+				address,
+				count: 1000,
+				distance: 10,
+				seed: expect.any(String),
+			});
+		});
+	});
+
+	describe('hasSeed', () => {
+		let address: string;
+		let address2: string;
+
+		beforeEach(async () => {
+			address = genesisDelegates.delegates[0].address;
+			const seed = genesisDelegates.delegates[0].hashOnion.hashes[1];
+			const count = 1000;
+			const distance = 10;
+			address2 = genesisDelegates.delegates[1].address;
+
+			await randomEndpoint.setHashOnion({ ...context, params: { address, seed, count, distance } });
+			await randomEndpoint.setHashOnion({
+				...context,
+				params: { address: address2, count, distance },
+			});
+
+			const usedHashOnionStore = randomEndpoint['offchainStores'].get(UsedHashOnionsStore);
+			await usedHashOnionStore.set(context, STORE_PREFIX_USED_HASH_ONION, {
+				usedHashOnions: [{ address: Buffer.from(address, 'hex'), count: 20, height: 2121 }],
+			});
+		});
+
+		it('should return hasSeed false with remaing 0 if hashOnion does not exist', async () => {
+			const hasSeed = await randomEndpoint.hasHashOnion({
```

Comment:
```suggestion
			const hasHashOnion = await randomEndpoint.hasHashOnion({
```
same for other tests